app/controllers/admin/site_settings_controller.rb

```diff
@@ -7,21 +7,24 @@ class Admin::SiteSettingsController < Admin::AdminController
 
   def index
     params.permit(:categories, :plugin)
+    params.permit(:filter_names, [])
 
     render_json_dump(
       site_settings:
         SiteSetting.all_settings(
           filter_categories: params[:categories],
           filter_plugin: params[:plugin],
+          filter_names: params[:filter_names],
+          include_locale_setting: params[:filter_names].blank?,
         ),
     )
   end
 
   def update
     params.require(:id)
     id = params[:id]
+    update_existing_users = params[:update_existing_user].present?
     value = params[id]
-    value.strip! if value.is_a?(String)
 
     new_setting_name =
       SiteSettings::DeprecatedSettings::SETTINGS.find do |old_name, new_name, override, _|
```

Comment:
I think we can keep this deprecated settings logic here because it's specific to this endpoint and IMO not something we need in other places/endpoints in the app. I believe we added this deprecated setting handling to support third-party API consumers that were calling this endpoint on settings that had been renamed and wanted the settings to continue to work.

Reply:
Okay that's fair, let's leave it for now :+1: